app/src/main/java/net/squanchy/tweets/service/TweetModelConverter.java

```diff
@@ -120,6 +114,25 @@ private int offsetFrom(int start, List<Integer> indices) {
         return Lists.map(photos, mediaEntity -> mediaEntity.mediaUrlHttps);
     }
 
+    private String displayableTextFor(Tweet tweet, Range displayTextRange, List<MediaEntity> photoUrls) {
+        Integer beginIndex = displayTextRange.start();
+        Integer endIndex = displayTextRange.end();
+        String displayableText = tweet.text.substring(beginIndex, endIndex);
+        if (photoUrls != null) {
+            displayableText = removePhotoUrl(displayableText, photoUrls);
+        }
+        return displayableText;
+    }
+
+    private String removePhotoUrl(String content, List<MediaEntity> photoUrls) {
+        for (MediaEntity url : photoUrls) {
+            if (content.contains(url.url)) {
+                content = content.replace(url.url, "");
```

Comment:
What if this affects other tags? If the image is mid-way through the tweet this might mess up all the spans!